### ISSUE TITLE
Automatic update of links is inconsistent when renaming files

### ISSUE THREAD

**Issue by tjhayes** [Zettlr/Zettlr#4989]
### Description

What I expect:
- after renaming a file, all links to that filename should update automatically

What I experience:
- after renaming a file, only some of the links to that file are automatically updated to the new filename. 
- some links are not updated. These become dead links to the old filename.

Zettlr Settings:
- I have Autosave set to Immediately

### Reproducing

(1) Create a set of files:
- Test Index.md
- Test 1.md
- Test 2.md
- Test 3.md

(2) In each file, create links to all of these files:
- [[Test Index]]
- [[Test 1]]
- [[Test 2]]
- [[Test 3]]

(3) Rename one of the files. For instance, rename "Test Index.md" to "My Test Index.md". 

(4) When Zettlr asks if you want to rename the links, click Yes.

(5) Look at the list of links in each file. Most of the files will have this list of links, showing the updated filename for the index file:
- [[My Test Index]]
- [[Test 1]]
- [[Test 2]]
- [[Test 3]]

(6) However, in one or more files, for example "Test 3.md", my list of links now looks like:
- [[Test Index]]
- [[Test 1]]
- [[Test 2]]
- [[Test 3]]

(7) For some reason, the link to the index [[Test Index]] was not renamed automatically to [[My Test Index]].

(8) I also confirmed this same behavior when I renamed "Test 1.md" to be "Test 1a.md". The links to that file were not all updated to be [[Test 1a]]. Some of the links still remain as [[Test 1]].

### Zettlr Version

Stable (most recent version)

### Installation Method

From the Website or GitHub

### Your Platform

- [ ] Windows
- [X] macOS
- [ ] Linux

### Architecture

- [ ] Intel 64bit
- [X] ARM 64bit

### Operating System Version

Sonoma 14.3.1

**Comment by nathanlesage** [comment]
Can you make out a pattern in which links get renamed and which do not? Also, could you please try out today's nightly as soon as it comes out and see if the issue persists there? https://nightly.zettlr.com

**Comment by tjhayes** [comment]
I tried to find a pattern about which links get renamed and which do not. I was able to reproduce this inconsistent link renaming, but then realized that I was actually running into a problem with the editor not re-loading or refreshing correctly, as mentioned [here:](https://github.com/Zettlr/Zettlr/issues/4896#issuecomment-1924896305) This editor display or refresh glitch was making it look like the links were not getting renamed automatically. However, if I close and reopen Zettlr, the links all look fine, and were indeed correctly renamed when the filenames changed.
